public/app.js

```diff
@@ -0,0 +1,61 @@
+// GLOBAL APP CONTROLLER
+class Controller {
+    static setupEventListeners() {
+        let DOMstrings = UIController.getDOMstrings();
+        document.querySelector(DOMstrings.logoutBtn).addEventListener('click', Controller.ctrlLogoutDashboard);
+        document.querySelector(DOMstrings.codingProgressLinkSelector).addEventListener('click', Controller.navigateToCodingProgress);
+        document.querySelector(DOMstrings.projectMenu).addEventListener('click', Controller.navigateToSelectedProject);          
+    };
+
+    static resetDashboard() {
+        let DOMstrings = UIController.getDOMstrings();
+        document.querySelector(DOMstrings.codingProgressContainer).innerHTML = "";
+        document.querySelector(DOMstrings.graphContainer).innerHTML = "";
+    }
+    
+    static ctrlLogoutDashboard() {
+        AuthController.logout()
+    };
+
+    static navigateToCodingProgress(e) {
+        Controller.resetDashboard()
+        if(e.target && e.target.nodeName == "A") {
+            // Add the coding progress section to the UI
+            UIController.addCodingProgressSection();
+            // Get data for coding progress table
+            DataController.watchCodingProgress(UIController.update_progress_ui);
+        }
+    };
+    
+    static navigateToSelectedProject(e) {
+        if(e.target && e.target.nodeName == "A") {
+            Controller.resetDashboard()
+            console.log(e.target.innerText)
+            let project = e.target.innerText
+             // Add the graphs container to the UI
+            UIController.addGraphs(project);
+            // Update and show the Graphs
+            DataController.watchProjectData(project, GraphController.update_graphs);
+        }
+    };  
+
+    static init() {
+        console.log('Application has started.');
+        // authorize user
+        AuthController.getUser();
+        // set up event listeners
+        Controller.setupEventListeners();
+        // Add the dropdown menu to the UI
+        DataController.watchActiveProjects(UIController.addDropdownMenu);
+        // Add the coding progress section to the UI
+        UIController.addCodingProgressSection();
+        // Get data for coding progress table
+        DataController.watchCodingProgress(UIController.update_progress_ui);
+    } 
+} 
+// initialize firestore
+const mediadb = firebase.firestore();
+const settings = { timestampsInSnapshots: true };
```

Comment:
In researching what this does I've discovered it is deprecated and can be removed:
https://firebase.google.com/docs/reference/js/firebase.firestore.Settings.html#timestampsinsnapshots

Reply:
I'm getting this warning after removing the setting which reverts me to adding the setting. 
![firestore](https://user-images.githubusercontent.com/39700595/71956754-05ae0900-31fd-11ea-9525-1f8a93fd18fb.png)
Also we are using the timestamps as suggested in the documentation.

Reply:
Ah, I suspect that's because we're using an old version of the client library. Maybe turn this into an issue instead

Reply:
Okay.

Reply:
Opened issue #128